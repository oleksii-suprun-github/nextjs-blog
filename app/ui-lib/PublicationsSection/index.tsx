import { ReactNode } from 'react';

function PublicationsSection({ children }: { children: ReactNode }) {
  return <div className="mb-24">{children}</div>;
}

export default PublicationsSection;
