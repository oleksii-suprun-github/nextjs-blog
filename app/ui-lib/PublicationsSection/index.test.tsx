import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import PublicationsSection from '.';

describe('PublicationsSection Component', () => {
  it('should render correctly and match snapshot', () => {
    const { asFragment } = render(
      <PublicationsSection>
        <div>Sample Content For Test</div>
      </PublicationsSection>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    render(
      <PublicationsSection>
        <div>Sample Content For Test</div>
      </PublicationsSection>,
    );

    const childElement = screen.getByText('Sample Content For Test');
    expect(childElement).toBeInTheDocument();
  });
});
