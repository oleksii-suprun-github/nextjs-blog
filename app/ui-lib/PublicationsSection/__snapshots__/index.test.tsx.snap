// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PublicationsSection Component > should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-24"
  >
    <div>
      Sample Content For Test
    </div>
  </div>
</DocumentFragment>
`;
